apps/passport-server/src/services/telegramService.ts

```diff
@@ -193,6 +205,82 @@ export class TelegramService {
         this.rollbarService?.reportError(e);
       }
     });
+
+    // The "setup" command helps prospective event organizer to collect necessary information.
+    // Currently, this just prints the ID of the channel.
+    this.bot.command("setup", async (ctx) => {
+      if (ctx.chat?.type === "private") {
+        await ctx.reply(
+          "To get you started, can you please add me as an admin to the telegram channel associated with your event? Once you are done, please ping me again with /setup in the channel."
+        );
+        return;
+      }
+
+      const admins = await ctx.getChatAdministrators();
+      const isAdmin = admins.some(
+        (admin) => admin.user.id === this.bot.botInfo.id
+      );
+      if (!isAdmin) {
+        await ctx.reply(
+          "Please add me as an admin to the telegram channel associated with your event."
+        );
+        return;
+      }
+
+      await ctx.reply("Your telegram channel id is " + ctx.chat.id);
+    });
+
+    // The  "link" command is a dev utility for associating the channel Id with a given event.
+    this.bot.command("link", async (ctx) => {
+      if (ctx.chat?.type === "private") {
+        await ctx.reply(
+          "To get you started, can you please add me as an admin to the telegram channel associated with your event? Once you are done, please ping me again with /setup in the channel."
+        );
+        return;
+      }
+
+      const admins = await ctx.getChatAdministrators();
+      const isAdmin = admins.some(
+        (admin) => admin.user.id === this.bot.botInfo.id
+      );
+      if (!isAdmin) {
+        await ctx.reply(
+          "Please add me as an admin to the telegram channel associated with your event."
+        );
+        return;
+      }
+      const channelId = ctx.chat.id;
+      // Must match the hard coded argument in event:dev command in package.json
+      const TEST_EVENT_NAME = "localTest";
+
+      try {
+        const eventInfo = await fetchPretixEventInfoByName(
+          this.context.dbPool,
+          TEST_EVENT_NAME
+        );
+
+        if (eventInfo?.pretix_events_config_id) {
+          await insertTelegramEvent(
+            this.context.dbPool,
+            eventInfo.pretix_events_config_id,
+            channelId
+          );
+          logger(
+            `[TELEGRAM] linked event ${TEST_EVENT_NAME} to group ${ctx.chat.title}`
+          );
+          await ctx.reply(
+            `Linked ${ctx.chat.title} (id: ${channelId}) with event ${TEST_EVENT_NAME}`
+          );
+        }
+        if (!eventInfo)
```

Comment:
nit: personally find it more readable to handle this case upfront before `eventInfo?.pretix_events_config_id` which could then be written as `eventInfo.pretix_events_config_id` or even eliminate the check if our API is trust worthy

Reply:
Closed by e622f6ea